### PR TITLE
Issue #SB-10796 feat: Ad-hoc job scripts

### PR DESCRIPTION
**Pull request by sunilpes**
scripts:
1. Cassandra to Redis indexer 
2. Elasticsearch to Redis indexer
3. Elasticsearch to Cloud uploader

### CODE REVIEW

File: druid/adhoc-jobs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        <dependency>
+            <groupId>com.typesafe</groupId>
+            <artifactId>config</artifactId>
+            <version>1.3.3</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.elasticsearch/elasticsearch-spark-20 -->
+        <dependency>
+            <groupId>org.elasticsearch</groupId>
+            <artifactId>elasticsearch-spark-20_2.11</artifactId>
+            <version>7.0.0-alpha2</version>
```
Comment (anandp504) [inline druid/adhoc-jobs/pom.xml]: Please use concrete versions and not alpha versions of a library.

File: druid/adhoc-jobs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/com.redislabs/spark-redis -->
+        <dependency>
+            <groupId>com.redislabs</groupId>
+            <artifactId>spark-redis</artifactId>
+            <version>2.3.1-M3</version>
```
Comment (anandp504) [inline druid/adhoc-jobs/pom.xml]: Is this milestone version of the spark-redis library stable?

File: druid/adhoc-jobs/pom.xml [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+    <groupId>org.ekstep</groupId>
+    <artifactId>adhoc-jobs</artifactId>
+    <version>1.0</version>
+
+    <properties>
+        <spark.version>2.0.1</spark.version>
+        <scala.version>2.11.8</scala.version>
+        <scala.maj.version>2.11</scala.maj.version>
+        <scoverage.plugin.version>1.1.1</scoverage.plugin.version>
```
Comment (anandp504) [inline druid/adhoc-jobs/pom.xml]: Are we using the scoverage plugin to generate code coverage? If not, please remove the dependency.

File: druid/adhoc-jobs/dependency-reduced-pom.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.ekstep</groupId>
+  <artifactId>adhoc-jobs</artifactId>
+  <version>1.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (sunilpes) [reply druid/adhoc-jobs/dependency-reduced-pom.xml]: done

File: druid/adhoc-jobs/src/main/scala/org/ekstep/analytics/jobs/CassandraToRedisIndexer.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        val sc = new SparkContext(conf)
+
+        val userKeyspace = config.getString("cassandra.user.keyspace")
+        val userTableName = config.getString("cassandra.user.table")
+        val redisKeyProperty: String = config.getString("redis.user.index.source.key")
+
+        val userData = sc.cassandraTable(userKeyspace, userTableName)
+        val userMap = userData.map(f => f.toMap)
+        val mappedData = userMap.map { obj =>
+            (obj(redisKeyProperty).asInstanceOf[String] , JSONUtils.serialize(obj))
```
[the remark this answers is not in the document]
Comment (sunilpes) [reply druid/adhoc-jobs/src/main/scala/org/ekstep/analytics/jobs/CassandraToRedisIndexer.scala]: done

File: druid/adhoc-jobs/pom.xml (hunk 1 above)
Comment (sunilpes) [reply druid/adhoc-jobs/pom.xml]: done

File: druid/adhoc-jobs/pom.xml (hunk 2 above)
Comment (sunilpes) [reply druid/adhoc-jobs/pom.xml]: tried with older version. But it was throwing error. Now switched to RC1 version

File: druid/adhoc-jobs/pom.xml (hunk 3 above)
Comment (sunilpes) [reply druid/adhoc-jobs/pom.xml]: done

**Pull request merged**